include/wally_elements.h

```diff
@@ -37,6 +42,28 @@ WALLY_CORE_API int wally_asset_generator_from_bytes(
     unsigned char *bytes_out,
     size_t len);
 
+/**
+ * Generate a rangeproof nonce hash using SHA256(ECDH(pub_key, priv_key).
+ *
+ * :param pub_key: Public blinding key.
```

Comment:
The shared secret used to encrypt data in the rangeproof is computed both ways
```
sha256(ecdh(receiver_public_blinding_key, sender_private_ephemeral_key)) == sha256(ecdh(sender_public_ephemeral_key, receiver_private_blinding_key))
```
I think the api documentation should reflect it, instead of describing only the sender way.

Also using `shared_nonce` instead of `nonce_hash` might be better (there is the `nonce` field but this is not simply its hash). But we might leave this name for consistency.

Reply:
Updated docs, left the name to keep what little consistency there currently is.